alchemist-web-renderer/src/jvmMain/kotlin/it/unibo/alchemist/server/state/ServerState.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2010-2022, Danilo Pianini and contributors
+ * listed, for each module, in the respective subproject's build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.server.state
+
+import it.unibo.alchemist.common.model.surrogate.EnvironmentSurrogate
+import it.unibo.alchemist.common.model.surrogate.PositionSurrogate
+import it.unibo.alchemist.common.state.CommonState
+import it.unibo.alchemist.core.interfaces.Simulation
+import it.unibo.alchemist.server.state.reducers.environmentSurrogateReducer
+import it.unibo.alchemist.server.state.reducers.simulationReducer
+
+/**
+ * The state of the server.
```

Comment:
@AngeloFilaseta can you document this class better, linking to the pattern and explaining the rationale behind the singleton state, like you did in the discussion comment?

Reply:
Perfect, I'll do the same thing for the ClientState.

Reply:
OK. Please resolve this conversation once done, and the PR will automerge.